core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -176,11 +179,13 @@ trait Endpoint[A] { self =>
       final def apply(input: Input): Endpoint.Result[O] = self(input) match {
         case a @ EndpointResult.Matched(_, _, _) => other(a.rem) match {
           case b @ EndpointResult.Matched(_, _, _) =>
-            EndpointResult.Matched(
-              b.rem,
-              a.trc.concat(b.trc),
-              a.out.liftToTry.product(b.out.liftToTry).flatMapF(this)
-            )
+            {
+              EndpointResult.Matched(
+                b.rem,
+                a.trc.concat(b.trc),
+                a.out.liftToTry.join(b.out.liftToTry).map( this).flatMap(f => Task.async(f))
```

Comment:
Can we move `Task.async` wrapping under `def apply` so we can keep `flatMap(this)` in place?